kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -60,12 +59,14 @@ public S3Storage(S3SinkConnectorConfig conf, String url) {
   }
 
   public AmazonS3 newS3Client(S3SinkConnectorConfig config) {
+    String proxyHost = config.getString(S3_PROXY_HOST);
+    int proxyPort = config.getInt(S3_PROXY_PORT);
     AmazonS3ClientBuilder builder = AmazonS3ClientBuilder.standard()
                                         .withAccelerateModeEnabled(config.getBoolean(WAN_MODE_CONFIG))
                                         .withPathStyleAccessEnabled(true)
                                         .withCredentials(config.getCredentialsProvider())
                                         .withClientConfiguration(
-                                            PredefinedClientConfigurations.defaultConfig()
+                                            PredefinedClientConfigurations.defaultConfig().withProxyHost(proxyHost).withProxyPort(proxyPort)
```

Comment:
I think you need to make this conditional on whether the values are actually set or not -- not all users will be using a proxy, so we can't always call these methods.